### PR TITLE
US36658 : GO SDK APIs for DHCP Server

### PR DESCRIPTION
**Pull request by Pushpalatha555**
validated the APIs

### CODE REVIEW

File: pkg/client/dhcp_server.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+)
+
+type DhcpServerAPIService struct {
+	Client APIClientHandler
+	Cfg    Configuration
+}
+
+func (lb *DhcpServerAPIService) CreateDhcpServer(
+	ctx context.Context,
+	serverID int,
```
Comment (manjunath-batakurki) [inline pkg/client/dhcp_server.go:21]: Can you please add compatible version ?

File: pkg/models/dhcp_server.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	ID              int         `json:"-" tf:"id,computed"`
+	NetworkServerID int         `json:"networkServerId" tf:"network_server_id,computed"`
+	Name            string      `json:"name" tf:"name"`
+	LeaseTime       int         `json:"leaseTime" tf:"lease_time"`
+	ServerIPAddress string      `json:"serverIpAddress" tf:"server_address"`
+	Config          *DhcpConfig `json:"config" tf:"config,sub"`
+}
+
+type DhcpConfig struct {
+	EdgeCluster string `json:"edgeCluster"  tf:"edge_cluster"`
```
Comment (manjunath-batakurki) [inline pkg/models/dhcp_server.go:22]: i think you need to pass the edge cluster provider id here. Hope you have required data sources created for it.

**Review by manjunath-batakurki — COMMENTED** [review]
few minor comments. changes looks good

**Review by manjunath-batakurki — COMMENTED** [review]
there is a merge conflicts, please resolve them as well

File: pkg/models/dhcp_server.go (hunk 2 above)
Comment (Pushpalatha555) [reply pkg/models/dhcp_server.go:22]: yes its der

File: pkg/client/dhcp_server.go (hunk 1 above)
Comment (Pushpalatha555) [reply pkg/client/dhcp_server.go:23]: updated the version

**Review by manjunath-batakurki — APPROVED** [review]

**Review by ravikumar29524 — APPROVED** [review]

**Pull request merged**